microservices-connector/cmd/router/main.go

```diff
@@ -477,6 +479,109 @@ func mcGraphHandler(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+func mcDataHandler(w http.ResponseWriter, r *http.Request) {
+	defaultNode := mcGraph.Spec.Nodes[defaultNodeName]
+	for i := range defaultNode.Steps {
+		step := &defaultNode.Steps[i]
+		if DataPrep == step.StepName {
```

Comment:
There should be only one dataprep service,   so we can break out the loop if we found it

Reply:
this will get us past the dataprep hump if we break out I like the full pass through in case there are other endpoints. Perhaps multiple different datapreps for different kinds of data - audio, video, images, text, web search, some agent. May be even different embedding services with their own vector database paths.

Also do steps have some description text, so that when someone queries for these endpoints they know what to use for what, like "query path", "data upload path" or something better.

Reply:
Hi @irisdingbj, I think we need to consider that there are multiple dataprep microservices in Pipeline, such as in switch mode. So I suggest that we can keep this logic, does it make sense?

Reply:
Hi @mkbhanda, the data formats supported by dataprep are only text format, such as txt, htm, md, etc. And for audio, video, images format, especially for video, we may need more time for supporting them because we need to handle the media stream in router. So suggest that we can just support text format as well, does it make sense?

Reply:
And some additional information, I handle all file in binary format, so in theories, it can handle all these kind of file format, but we need to handle the stream properly.

Reply:
Yeah, this should be. So I original support the extra request Header of `SERVICE_NAME`, but remove it because I want to make the same format with `dataprep` microservice